apisix/plugins/jwt-auth.lua

```diff
@@ -274,32 +234,16 @@ end
 local function get_rsa_or_ecdsa_keypair(conf, consumer_name)
     local public_key = conf.public_key
     local private_key = conf.private_key
-    -- if keys are present in conf, no need to query vault (fallback)
+
     if public_key and private_key then
         return public_key, private_key
+    elseif public_key and not private_key then
+        return public_key, nil, "missing private key"
```

Comment:
Will it be a problem in https://github.com/shreemaan-abhishek/apisix/blob/09f6c5971176b6fd60b9c090aacd1c556c072d1d/apisix/plugins/jwt-auth.lua#L291?

Reply:
Should we add a nil check there? 🤔

Reply:
should we add a nil check there?

Reply:
https://github.com/apache/apisix/blob/9c899b87648039e2c6c73b087d627aea2df90f55/apisix/plugins/jwt-auth.lua#L347
Here we check if public_key is nil to detect the err.

Reply:
I am not quite sure what you are trying to say, could you please elaborate? 😅